### ISSUE TITLE
SFTP support for `about` command

### ISSUE THREAD

**Issue by atruskie** [ncw/rclone#3107]
#### What is your current rclone version (output from `rclone version`)?

```
rclone v1.46
- os/arch: windows/amd64
- go version: go1.11
```

#### What problem are you are trying to solve?

I'd love to use the `about` command for the SFTP remote type. Currently:

```
.\rclone about remote:
2019/04/10 14:59:54 Failed to about: sftp://user@remote: doesn't support about
```

#### How do you think rclone should be changed to solve that?

Similarly to how rclone uses md5 and sha1sum commands if it can to generate hashes, it would be great if it could try and use the `df` command to query space for the current device.

Since a remote system can have multiple disks, i'd suggest returning the information for whatever disk is currently mounted to the target remote folder. This can be done by supplying a path argument to `df`.

Not all of the information for command would be available, but `about` doesn't guarantee all info is available.

**Comment by ncw** [comment]
That's a great idea!  Do you want to work on it?

**Comment by atruskie** [comment]
I'd like to, if I find the time, but as with my other post, I don't know Go.

**Comment by gary-kim** [comment]
I'll see if I can implement this.
This is my first time using Go so I'm going to be pulling a lot of cues from other parts of `rclone`. I'm sorry if it ends up being messy.

**Comment by ncw** [comment]
> I'll see if I can implement this.
> This is my first time using Go so I'm going to be pulling a lot of cues from other parts of `rclone`. I'm sorry if it ends up being messy.

That would be great :-)  Here are some hints

What you'll have to do...
- write a parser for the output of `df`
- write a unit test for that
- Implement About - this could should call `df` and parse the output

Open questions
- should `df` be called with a parameter? I think so but I'm not sure exactly what! Maybe `df root` where root is the defined `f.root`.
- should `df` be called with any flags? We need to be careful with cross platform stuff here... Note that GNU df outputs in 1k blocks but POSIX df outputs in blocks of 512... So it might be nice to collect some examples from Linux/MacOS/FreeBSD for the unit tests.
- Should we detect whether df is available (like we do for md5sum/sha1sum) at Fs initialisation then remove the About call?

Use this bit of code as a guide for calling `df` and getting the output

https://github.com/ncw/rclone/blob/aba15f11d85dbb0b137f4b9f8d009d4c6fd7e47a/backend/sftp/sftp.go#L866-L881

And here is an implemention of `About` for you to look at

https://github.com/ncw/rclone/blob/aba15f11d85dbb0b137f4b9f8d009d4c6fd7e47a/backend/drive/drive.go#L1955-L1981

**Comment by ncw** [comment]
I've merged this to master now which means it will be in [the latest beta](https://beta.rclone.org) in 15-30 mins and released in v1.48

Thank you to @gary-kim for implementing the feature.
